cli/command/node/completion.go

```diff
@@ -0,0 +1,36 @@
+package node
+
+import (
+	"os"
+
+	"github.com/docker/cli/cli/command/completion"
+	"github.com/docker/docker/api/types"
+	"github.com/spf13/cobra"
+)
+
+// completeNodeNames offers completion for swarm node (host)names and optional IDs.
+// By default, only names are returned.
+// Set DOCKER_COMPLETION_SHOW_NODE_IDS=yes to also complete IDs.
+//
+// TODO(thaJeztah): add support for filters.
+func completeNodeNames(dockerCLI completion.APIClientProvider) completion.ValidArgsFn {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		list, err := dockerCLI.Client().NodeList(cmd.Context(), types.NodeListOptions{})
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		showIDs := os.Getenv("DOCKER_COMPLETION_SHOW_NODE_IDS") == "yes"
+		names := make([]string, 0, len(list))
```

Comment:
```suggestion
		names := make([]string, 0, len(list) + 1)
```
To account for the `self` magic keyword?

Reply:
Ah! Yes, I added that one later; list would also be 2 x len(list) if IDs are shown

Reply:
Yeah, I didn't want to complicate it more than needed to be (making the length conditional)). For fun, we could benchmark though if unconditionally allocating `2 * len (+1)` would be better or worse than starting with `len (+1)`, then growing?

Reply:
All irrelevant really for this code, as the API call is much more expensive than just this 😂 (but fun to know!)